### PR TITLE
Update cookiecutter, restructure library

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
  - This PR fixes #xyz
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] CHANGES.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Updates the cookiecutter to the latest commit.
* The `ruff` linter has been adjusted to render the `flake8-alphabetize` dependency redundant (and removed).
* Several `noqa` and `fixme` statements have been added to pass additional linting checks for now.

### Does this PR introduce a breaking change?

Yes. 
* The structure of the package is slightly modified from a flat layout to a src layout. (https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/)
* `CHANGES.rst` is now `CHANGELOG.rst`. (https://keepachangelog.com/en/1.1.0/#frequently-asked-questions)
* Bumping a release version will trigger changes in the `CHANGELOG.rst` file. (https://github.com/Ouranosinc/cookiecutter-pypackage/issues/41)

### Other information:

~This does not yet integrate changes from https://github.com/Ouranosinc/cookiecutter-pypackage/pull/50 (but it should before merging)~

We should perform a release after this and then see about proposing the package to `conda-forge`!

### CODE REVIEW

File: .cruft.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
       "create_author_file": "y",
       "open_source_license": "MIT license",
       "generated_with_cruft": "y",
+      "__gh_slug": "https://github.com/sebastienlanglois/xdatasets",
```
Comment (RondeauG) [inline .cruft.json:25]:
```suggestion
      "__gh_slug": "https://github.com/hydrologie/xdatasets",
```

**Review by RondeauG — APPROVED** [review]
Cookie-cutter needs to be fast-forwarded, but otherwise LGTM!

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -58,211 +56,173 @@ Get Started!
...
+            git clone git@github.com:your_name_here/xdatasets.git
 
-#. Create a branch for local development::
+    * If you are a ``xdatasets`` collaborator, clone the ``xdatasets`` repo directly.
 
-    $ git checkout -b name-of-your-bugfix-or-feature
+        .. code-block:: console
 
-  Now you can make your changes locally.
+            git clone git@github.com:``sebastienlanglois``/xdatasets.git
```
Comment (RondeauG) [inline CONTRIBUTING.rst:84]:
```suggestion
            git clone git@github.com:hydrologie/xdatasets.git
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -58,211 +56,173 @@ Get Started!
...
+#. Install your local copy into a development environment. You can create a new Anaconda development environment with:
 
-    $ python -m pytest
-    # Or, to run multiple build tests
-    $ tox
+    .. code-block:: console
 
-#. Commit your changes and push your branch to GitHub::
+        conda env create -f environment-dev.yml
+        conda activate xdatasets-dev
```
Comment (RondeauG) [inline CONTRIBUTING.rst:91]:
```suggestion
        conda activate xdatasets
```

Comment (RondeauG) [reply CONTRIBUTING.rst:91]: The environment here is named `xdatasets` in the `environment-dev` file.

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -58,211 +56,173 @@ Get Started!
...
-    $ /opt/python/cp39-cp39m/bin/python -m build --sdist --wheel
+To get ``black``, ``isort``, ``blackdoc``, ``ruff``, and ``flake8`` (with plugin ``flake8-rst-docstrings``) simply install them with ``pip`` (or ``conda``) into your environment.
 
-This will then place two files in `xdatasets/dist/` ("xdatasets-1.2.3-py3-none-any.whl" and "xdatasets-1.2.3.tar.gz").
-We can now leave our docker container (`$ exit`) and continue with uploading the files to PyPI::
+Code of Conduct
+---------------
 
-    $ twine upload dist/*
+Please note that this project is released with a `Contributor Code of Conduct <https://github.com/sebastienlanglois/xdatasets/blob/main/CODE_OF_CONDUCT.md>`_.
```
Comment (RondeauG) [inline CONTRIBUTING.rst:227]:
```suggestion
Please note that this project is released with a `Contributor Code of Conduct <https://github.com/hydrologie/xdatasets/blob/main/CODE_OF_CONDUCT.md>`_.
```

File: docs/installation.rst [hunk 5]
```diff
@@ -4,48 +4,94 @@
...
+    .. code-block:: console
+
+        python -m pip install .
+
+    ..
+
+        .. code-block:: console
+
+            conda env create -f environment-dev.yml
+            conda activate xdatasets-dev
```
Comment (RondeauG) [inline docs/installation.rst:64]:
```suggestion
            conda activate xdatasets
```

File: docs/installation.rst [hunk 6]
```diff
@@ -4,48 +4,94 @@
...
+
+    ..
+        .. code-block:: console
 
-    $ $ python -m pip install .
+            git fetch
+            git checkout main
+            git pull origin main
+            conda env update -n xdatasets-dev -f environment-dev.yml
+            conda activate xdatasets-dev
```
Comment (RondeauG) [inline docs/installation.rst:92]:
```suggestion
            conda env update -n xdatasets -f environment-dev.yml
            conda activate xdatasets
```

File: CONTRIBUTING.rst (hunk 3 above)
Comment (Zeitsperre) [reply CONTRIBUTING.rst:91]: Right. That's not the case everywhere. No need to push that change up.

File: .cruft.json (hunk 1 above)
Comment (RondeauG) [inline .cruft.json:25]:
```suggestion
      "__gh_slug": "https://github.com/hydrologie/xdatasets",
```

**Review by RondeauG — APPROVED** [review]

**Pull request merged**